webapp/src/dogma/common/components/table/DataTable.tsx

```diff
@@ -0,0 +1,150 @@
+import { DeleteIcon, TriangleDownIcon, TriangleUpIcon, ViewIcon } from '@chakra-ui/icons';
+import {
+  Button,
+  ButtonGroup,
+  chakra,
+  IconButton,
+  Popover,
+  PopoverArrow,
+  PopoverBody,
+  PopoverCloseButton,
+  PopoverContent,
+  PopoverFooter,
+  PopoverHeader,
+  PopoverTrigger,
+  Table,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr,
+  Wrap,
+  WrapItem,
+} from '@chakra-ui/react';
+import {
+  ColumnDef,
+  ColumnFiltersState,
+  flexRender,
+  getCoreRowModel,
+  getFilteredRowModel,
+  getSortedRowModel,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table';
+import { Filter } from 'dogma/common/components/table/Filter';
+import Link from 'next/link';
+import { useState } from 'react';
+
+export type DataTableProps<Data extends object> = {
+  data: Data[];
+  urlPrefix: string;
+  columns: ColumnDef<Data, any>[];
+};
+
+export const DataTable = <Data extends object>({ data, urlPrefix, columns }: DataTableProps<Data>) => {
```

Comment:
This component is more than just a data table. How about renaming `DataTable` into `DynamicDataTable` or something else?